apps/src/lib/kits/maker/toolkit.js

```diff
@@ -146,11 +146,20 @@ function shouldRunWithFakeBoard() {
 
 /**
  * Called when execution of the student app ends.
- * Resets the board state, disconnects and destroys the current board controller,
+ * Resets the board state
  * and puts maker UI back in a default state.
  */
 export function reset() {
   if (currentBoard) {
     currentBoard.reset();
   }
 }
+
+/**
+ * Called when the board disconnects
+ * Throw away reference to the currentBoard, so that next time we run
+ * we make a new board.
+ */
+export function disconnectBoard() {
+  currentBoard = null;
```

Comment:
This implementation does not reflect the disconnect logic you removed from `reset()` [in 28037](https://github.com/code-dot-org/code-dot-org/pull/28037/files#diff-c8149494be9671c23275e8bdbc4ea056L166).  For instance, you're not calling `currentBoard.destroy()` which performs some important teardown, even if the board is no longer reachable:

https://github.com/code-dot-org/code-dot-org/blob/f90bd8360531aec6137e4655d10040cf0f862694/apps/src/lib/kits/maker/CircuitPlaygroundBoard.js#L165-L171